src/utils/structuredData.ts

```diff
@@ -18,9 +18,9 @@ export const generateSeriesMetadata = (seriesPlaylist: Playlist) => {
   };
 };
 
-export const generateEpisodeJSONLD = (seriesPlaylist: Playlist, episode: PlaylistItem) => {
-  const episodeCanonical = `${window.location.origin}${episodeURL(episode, seriesPlaylist.feedid)}`;
-  const seriesMetadata = generateSeriesMetadata(seriesPlaylist);
+export const generateEpisodeJSONLD = (seriesPlaylist: Playlist, episode: PlaylistItem, seriesId: string | undefined, mediaId: string | undefined) => {
+  const episodeCanonical = `${window.location.origin}${episodeURL({ episode, playlistId: seriesPlaylist.feedid, mediaId, seriesId })}`;
```

Comment:
I think the usage of `seriesPlaylist.feedid` here is a bug. We don't want to capture the legacy series playlist, but rather the playlist for the shelf that the playlist was in, right?